yarn-project/prover-client/src/orchestrator/orchestrator.ts

```diff
@@ -138,16 +141,22 @@ export class ProvingOrchestrator implements EpochProver {
   /**
    * Starts off a new block
    * @param numTxs - The total number of transactions in the block.
+   * @param numTxsEffects - The total number of transaction effects in the block
    * @param globalVariables - The global variables for the block
    * @param l1ToL2Messages - The l1 to l2 messages for the block
    * @param verificationKeys - The private kernel verification keys
    * @returns A proving ticket, containing a promise notifying of proving completion
    */
-  @trackSpan('ProvingOrchestrator.startNewBlock', (numTxs, globalVariables) => ({
+  @trackSpan('ProvingOrchestrator.startNewBlock', (numTxs, _, globalVariables) => ({
     [Attributes.BLOCK_SIZE]: numTxs,
     [Attributes.BLOCK_NUMBER]: globalVariables.blockNumber.toNumber(),
   }))
-  public async startNewBlock(numTxs: number, globalVariables: GlobalVariables, l1ToL2Messages: Fr[]) {
+  public async startNewBlock(
+    numTxs: number,
+    numTxsEffects: number,
```

Comment:
Looks like it's inevitable that we have to know the size of all txEffects to start proving. I suggest that we:
- Remove `numTxs` and `numTxsEffects` here, so this method is only for initializing the BlockProvingState with the correct global variables and start states.
- Change the `addNewTx` method below to `addTxs`. Get the size of all txEffects to initialize SpongeBlob.
  - This will remove all the `toNumTxsEffects` and for loops in the tests.
- Don't send `processedTxHandler` to PublicProcessor anymore. Instead, the parent component gets the processed txs from PublicProcessor, and pass them to whatever it needs them for.
  - We only need to add an additional line to two components to add the txs, and we get to remove the ugly if statement in PublicProcessor and some undefined values passed to it.

Reply:
This sounds much better, thank you! I will tackle this in a couple of commits time as it's quite a big change. IIRC there are some cases where we don't add all the txs at once (e.g. `addNewTx`, do something else, `addNewTx`), I think it's ok to rework those to add all txs with the one call?

Reply:
Have reworked, lmk if it looks good!

Reply:
So clean and pretty now. Thanks for changing it!